base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -1493,6 +1535,21 @@ fn rewind_to_height<T: BlockchainBackend>(db: &mut T, height: u64) -> Result<Vec
     );
     db.write(txn)?;
 
+    // Update reorg info
+    let reorg_info = ReorgInfo {
```

Comment:
This seems to be a duplicate of the `add_block` logic. Are you sure this is not done twice?

Reply:
There are different pathways/edge cases catered for here, one of which are when the entire database a base node has assembled are reorged away.